addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -580,18 +575,49 @@ var FileWidget = SearchableMediaWidget.extend({
      * @returns {Promise}
      */
     async _addData() {
-        const files = this.$fileInput[0].files;
+        let files = this.$fileInput[0].files;
         if (!files.length) {
             // Case if the input is emptied, return resolved promise
             return;
         }
-        this.uploader = new UploadProgressToast(this, {
-            files: files,
-            resModel: this.options.res_model,
-            resId: this.options.res_id,
-            isImage: this.widgetType === 'image',
+
+        const uploadMutex = new concurrency.Mutex();
+
+        // Upload the smallest file first to block the user the least possible.
+        files = _.sortBy(files, 'size');
+        await this._setUpProgressToast(files);
+        _.each(files, (file, index) => {
+            // Upload one file at a time: no need to parallel as upload is
+            // limited by bandwidth.
+            uploadMutex.exec(async () => {
+                const dataURL = await utils.getDataURLFromFile(file);
+                const attachment = await this.uploader._rpcShowProgress({
```

Comment:
Should not call a private method :grimacing: Simply make it public maybe?

Reply:
indeed, rookie mistake here